src/components/PartnerBalances.tsx

```diff
@@ -20,25 +23,55 @@ function PartnerBalance({ label, balance, symbol }: PartnerBalanceProp) {
 }
 
 function PartnerBalances() {
-    const { state } = useStore();
+    const { state, dispatch } = useStore();
 
-    const { worker, collector, partners, token } = state;
+    const { token, provider, reloadPartners } = state;
+
+    const [feesReceiver, setFeesReceiver] = useState<Partner | undefined>(
+        undefined
+    );
+    const [partners, setPartners] = useState<Partner[]>([]);
+
+    const getPartnerBalance = async (address: string) => {
+        try {
+            const balance = await Utils.getTokenBalance(address, token!);
+            return { address, balance };
+        } catch (error) {
+            console.error(error);
+            return { address, balance: '-' };
+        }
+    };
+
+    const refreshPartners = useCallback(async () => {
+        if (provider && reloadPartners) {
+            const localPartners = await provider.getPartners();
+            const updatedBalances = await Promise.all(
+                localPartners.map((partner) => getPartnerBalance(partner))
+            );
+            const [newFeesReceiver, ...newPartners] = updatedBalances;
+            setFeesReceiver(newFeesReceiver);
+            setPartners(newPartners);
+        }
+    }, [token, reloadPartners]);
```

Comment:
I could include it and the only difference would be that when we change the provider it would perform a request, but changing the provider does not change the partners, thats why I did not include it.

Reply:
Thanks for the explanation, it makes sense not to include it 👍🏽